### PR TITLE
Submission

### PR DESCRIPTION
**Pull request by tylercorbett**

### CODE REVIEW

**Comment by tylercorbett** [comment]
PRAISE THE CODING GODS TRAVIS HAS PASSED

File: lib/app.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
```
Comment (acrankin) [inline lib/app.js]: You're missing some middleware in here. You're going to need to add the following: 
```
const connection = require('./middleware/connection');
const { handler } = require('./middleware/error');
const notFound = require('./middleware/notFound');
```

File: lib/app.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+const app = express();
+const tweets = require('./routes/tweets');
+const users = require('./routes/users');
+
+app.use(express.json());
+app.use('/tweets', tweets);
+app.use('/users', users);
+
```
Comment (acrankin) [inline lib/app.js]:
```suggestion
app.use(notFound);

app.use(handler);
```

File: lib/app.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+const app = express();
+const tweets = require('./routes/tweets');
+const users = require('./routes/users');
+
+app.use(express.json());
+app.use('/tweets', tweets);
```
Comment (acrankin) [inline lib/app.js]:
```suggestion
app.use('/tweets', connection, tweets);
```

File: lib/middleware/connection.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const state = require('mongoose/lib/connection/state');
```
Comment (acrankin) [inline lib/middleware/connection.js]: you need to require mongoose in this file

Comment (acrankin) [inline lib/middleware/connection.js]:
```suggestion
const state = require('mongoose/lib/connectionstate');
```

**Pull request merged**